jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java

```diff
@@ -744,7 +789,7 @@ private static long[] dmatrixsToHandles(DMatrix[] dmatrixs) {
   private void writeObject(java.io.ObjectOutputStream out) throws IOException {
     try {
       out.writeInt(version);
-      out.writeObject(this.toByteArray());
+      out.writeObject(this.toByteArray("json"));
```

Comment:
Is it possible that we don't make this change in this PR? We want to have consistent a default for all language bindings.

Reply:
We have to change it to `json` format, because the default format [SaveModel](https://github.com/dmlc/xgboost/blob/master/src/learner.cc#L1088-L1134) and [LoadModel](https://github.com/dmlc/xgboost/blob/master/src/learner.cc#L956-L1084) ignores the feature_name and feature_type fields. If we don't want to change to json format, we need to modify the SaveModel and LoadModel functions. Since the default format will be deprecated, changing format here is simple and this function is only used when serialization/deserialization between spark workers, I think we can just change it here.

Reply:
Thank you for sharing, let's change it to `ubj` instead, which is more efficient.

Reply:
Changed to `ubj` format, thank you.